app.py

```diff
@@ -388,10 +388,10 @@ def direct_entry():
 		return jsonify({'error': 'Sentence ID is empty.'})
 	db = getdb()
 	cur = db.execute(
-		'SELECT * FROM entries ORDER BY sentno ASC'
+		'SELECT id FROM entries ORDER BY sentno ASC'
 	)
 	entries = cur.fetchall()
-	existing_ids = {entry[0] for entry in entries} | {entry[3] for entry in QUEUE}
+	existing_ids = set(entries) | {entry[3] for entry in QUEUE}
```

Comment:
Won't each entry in `entries` be a tuple? Presumably you just want the string, so `entry[0]` is still important.

Reply:
Yes, that's right! Thanks for catching this. (Updated in https://github.com/nschneid/activedop/pull/113/commits/cf1974a69a3f68acb9baa33f34196cd65217ee63)

Reply:
You changed it to `set(entries[0])`, but why not the original—`{entry[0] for entry in entries}`?